src/testnet/helium/event_dispatcher.rs

```diff
@@ -0,0 +1,242 @@
+use std::collections::hash_map::Entry;
+use std::collections::{HashMap, HashSet};
+use std::thread::spawn;
+use std::net::{SocketAddr};
+use mio::tcp::TcpStream;
+use serde_json::json;
+use serde::Serialize;
+
+use vm::types::{Value, QualifiedContractIdentifier, AssetIdentifier};
+use burnchains::Txid;
+use chainstate::stacks::StacksBlock;
+use chainstate::stacks::events::{StacksTransactionReceipt, StacksTransactionEvent, STXEventType, FTEventType, NFTEventType};
+use chainstate::stacks::db::StacksHeaderInfo;
+use net::StacksMessageCodec;
+
+use super::config::{EventObserverConfig, EventKeyType};
+
+#[derive(Debug)]
+struct EventObserver {
+    sock_addr: SocketAddr
+}
+
+impl EventObserver {
+
+    pub fn new(address: &str, port: u16) -> EventObserver {
+        let sock_addr = SocketAddr::new(address.parse().unwrap(), port);
+        EventObserver { sock_addr }
+    }
+
+    pub fn send(&mut self, filtered_events: Vec<&(Txid, &StacksTransactionEvent)>, chain_tip: &StacksBlock, chain_tip_info: &StacksHeaderInfo, receipts: &Vec<StacksTransactionReceipt>) {
```

Comment:
Just so you're aware, this code implements both a _blocking_ `connect(2)` and a _blocking_ `send(2)`.  This means that a slow observer can cause the entire node to stall.  I don't see any documentation of this behavior anywhere.  Is this behavior intended?

Reply:
Can we address this in a separate PR with https://github.com/blockstack/stacks-blockchain/issues/1352 once your HTTP client is available in master?

Reply:
The HTTP client is already in master -- but either way, I thought that we had decided this was the behavior we wanted for the sidecar -- the core node should stall until the sidecar has acknowledged receipt of the data (by accepting the HTTP request, or in this PR's case, the send).

Reply:
Okay, sounds good :+1:  Just wanted to make sure we were all on the same page about it.